xmodule/modulestore/tests/test_mongo.py

```diff
@@ -812,10 +817,17 @@ def _check_delete_key_error(self, key):
             self.kvs.get(key)
         assert not self.kvs.has(key)
 
-    def test_delete(self):
-        yield (self._check_delete_key_error, KeyValueStore.Key(Scope.content, None, None, 'foo'))
-        yield (self._check_delete_default, KeyValueStore.Key(Scope.children, None, None, 'children'), [])
-        yield (self._check_delete_key_error, KeyValueStore.Key(Scope.settings, None, None, 'meta'))
+    def test_delete_key_error(self):
+        items = [
+            {"scope": Scope.content, "expected": "foo"},
+            {"scope": Scope.settings, "expected": "meta"},
+        ]
+        for item in items:
+            with self.subTest():
+                self._check_delete_key_error(KeyValueStore.Key(item["scope"], None, None, item["expected"]))
```

Comment:
Let's move the content of the `_check_delete_key_error` here. It's not used in other tests, so it does not make much sense to keep it separated.